test_key.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/sh
+
+if ! [ -f autograder/repo ]; then
+  echo 'There does not appear to be a configured autograder here.'
+  echo 'Have you run `./configure_autograder.sh` yet?'
+  exit 1
+fi
+
+sed "s;/root/.ssh/deploy_key;$(realpath autograder/deploy_key);" autograder/ssh_config > local_ssh_config
+
+if GIT_SSH_COMMAND="ssh -F local_ssh_config" git ls-remote -q --refs "$(cat autograder/repo)" HEAD; then
```

Comment:
What's the reason we can't use "ssh  -F autograder/ssh_config..." as opposed to making a copy of the ssh config?

Reply:
Because the copy of the ssh_config uses the identity file in autograder/deploy_key instead of the identity in /root/.ssh/id_rsa

Reply:
The reason we use -F with a config at all (instead of -i with an identity) is because ssh will fallback to the default config file if the key fails -- it'll prompt the user to login instead of reporting a problem with the key.